storage/src/iter.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+//! Internal utilities for working with iterators.
+
+/// Writes an iterator of displayable items to a writer, separated by a specified separator.
+///
+/// - If `limit` is `Some(0)`, it will only write the number of hidden items.
+/// - If `limit` is `Some(n)`, it will write at most `n` items, followed by a message
+///   indicating how many more items were not written.
+/// - If `limit` is `None`, it will write all items without any limit.
+///
+/// # Arguments
+/// - `writer`: The writer to which the items will be written.
+/// - `iter`: An iterator of items that implement `std::fmt::Display`.
+/// - `sep`: A separator that will be used between items.
+/// - `limit`: An optional limit on the number of items to write.
+///
+/// # Returns
+/// A `std::fmt::Result` indicating success or failure of the write operation.
+pub(crate) fn write_iter(
```

Comment:
Maybe `write_limited_with_sep`? Generally functions that end in `iter` return iterators, which this does not.